projects/easywsclient/easyws_fuzzer.cpp

```diff
@@ -0,0 +1,27 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <stddef.h>
+#include <stdint.h>
+#include <string.h>
+
+#include "easywsclient.hpp"
+#include <fuzzer/FuzzedDataProvider.h>
+
+extern "C" int LLVMFuzzerTestOneInput(const uint8_t* data, size_t size) {
+  FuzzedDataProvider stream(data, size);
+  const std::string str = stream.ConsumeRandomLengthString(size);
+  easywsclient::WebSocket::from_url(str);
```

Comment:
This seems like a really basic fuzz target. Are you planning on adding more interesting ones?
Also, do you have any idea if these sockets close themselves? Or is this leaking resources?

Reply:
This is a starting point for the owners or contributors of easywsclient, which is written in the description above. They will take it over and add more fuzz targets when they find time. The `easywsclient::WebSocket` closes itself.

Reply:
Why is it being checked into OSS-Fuzz instead of the upstream repo then?